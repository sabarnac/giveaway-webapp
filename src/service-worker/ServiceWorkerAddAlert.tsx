import React, { useState } from "react";
import { inject } from "mobx-react";
import { WithTranslation, withTranslation } from "react-i18next";
import ServiceWorkerAlertsConfig from "../store/config/ServiceWorkerAlertsConfig";
const SweetAlert = require("sweetalert2-react");

/**
 * Properties of the service worker add alert React component.
 */
interface ServiceWorkerAddAlertProps {
  /** @ignore The service worker alerts config. */
  serviceWorkerAlertsConfig?: ServiceWorkerAlertsConfig;
}

/**
 * React component for the service worker add alert.
 */
export default withTranslation()(
  inject("serviceWorkerAlertsConfig")(
    (props: ServiceWorkerAddAlertProps & WithTranslation): JSX.Element => {
      const [show, setShow] = useState(true);

      const title: string = props.t("serviceWorker.offlineMessage.title");
      const message: string = props.t("serviceWorker.offlineMessage.message");

      return (
        <SweetAlert.default
          show={show && props.serviceWorkerAlertsConfig!.added}
          title={title}
          text={message}
          onConfirm={() => setShow(false)}
        />
      );
    },
  ),
);
