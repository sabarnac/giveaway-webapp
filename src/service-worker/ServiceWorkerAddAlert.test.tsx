import React from "react";
import ServiceWorkerAddAlert from "./ServiceWorkerAddAlert";
import { create, ReactTestRenderer } from "react-test-renderer";

describe("ServiceWorkerAddAlert Component.", () => {
  it("Matches snapshot.", () => {
    const component: ReactTestRenderer = create(<ServiceWorkerAddAlert />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
