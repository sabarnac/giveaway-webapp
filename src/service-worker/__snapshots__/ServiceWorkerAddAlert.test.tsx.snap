// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceWorkerAddAlert Component. Matches snapshot. 1`] = `
<div
  onConfirm={[Function]}
  show={true}
  text="serviceWorker.offlineMessage.message"
  title="serviceWorker.offlineMessage.title"
/>
`;
